lms/models/lms_course.py

```diff
@@ -0,0 +1,84 @@
+"""Models to represent courses.
+
+These duplicate some of the information stored in Grouping and GroupingMembership, the main differences being:
+
+    - One LMSCourse row per course, not one per (application_instance_id, course) like in Grouping.
+    - One or more rows in LMSCourseApplicationInstance, one per install we have seen the course in.
+    - LMSCourse membership stores role information, GroupingMembership doesn't.
+"""
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Mapped, mapped_column
+
+from lms.db import Base
+from lms.models._mixins import CreatedUpdatedMixin
+
+
+class LMSCourse(CreatedUpdatedMixin, Base):
+    __tablename__ = "lms_course"
+    __table_args__ = (
+        # GUID and lti_context_id are unique together.
+        # lti_context_id identifies the course within the LMS system and we use GUID to identify each LMS instance
+        sa.UniqueConstraint("tool_consumer_instance_guid", "lti_context_id"),
+    )
+
+    id: Mapped[int] = mapped_column(autoincrement=True, primary_key=True)
+
+    tool_consumer_instance_guid: Mapped[str | None] = mapped_column(
+        sa.UnicodeText, index=True
```

Comment:
Do you actually need the `sa.UnicodeText` when already using `Mapped[str]`? I think SQLAlchemy chooses the right type automatically?

Reply:
Yeah, I think we can settle on the type chosen by SQLAlchemy for str. 

There's many types it could chose String, Unicode, UnicdeText, it chooses the equivalent of String/Unicode which we use in many columns. I think we can settle on that.